### PR TITLE
Add group identity support

### PR DESCRIPTION
**Pull request by Powersource**
for https://github.com/ssbc/ssb-uri-spec/issues/18

### CODE REVIEW

**Comment by Powersource** [comment]
@staltz @mixmix

**Comment by Powersource** [comment]
saw that cloaked links didn't work to regex so added that in the first commit

**Comment by Powersource** [comment]
added to readme

File: src/index.ts [hunk 1]
```diff
@@ -18,7 +18,10 @@ type MessageTF =
 type BlobTF = ['blob', 'classic'] | ['blob', 'sha256'];
 type AddressTF = ['address', 'multiserver'];
 type EncryptionKeyTF = ['encryption-key', 'box2-dm-dh'];
-type IdentityTF = ['identity', 'po-box'] | ['identity', 'fusion'];
+type IdentityTF =
+  | ['identity', 'po-box']
+  | ['identity', 'group']
+  | ['identity', 'fusion'];
```
Comment (staltz) [inline src/index.ts:24]: Comment that does not block this PR: I just realized that ssb-bfe-spec doesn't have fusion, should we add it there?

File: test/compose.js [hunk 2]
```diff
@@ -27,6 +27,38 @@ test('decompose() a cloaked message', (t) => {
...
+  });
+  t.equals(uri, fixtures.identity.uri3);
+
+  t.end();
+});
+
+test('decompose() identities', (t) => {
+  ssbUri.decompose(fixtures.identity.uri);
+  ssbUri.decompose(fixtures.identity.uri2);
+  ssbUri.decompose(fixtures.identity.uri3);
```
Comment (staltz) [inline test/compose.js:58]: There's nothing being asserted here, perhaps you could just do a `t.equals` that the `.format` is po-box, fusion, and group

**Review by staltz — CHANGES_REQUESTED** [review]
Just one thing, then we can merge.

File: test/compose.js (hunk 2 above)
Comment (Powersource) [reply test/compose.js:58]: it does throw when it errors. the test was failing before i made the function work iirc

File: src/index.ts (hunk 1 above)
Comment (Powersource) [reply src/index.ts:24]: problem for another day lol

File: test/compose.js (hunk 2 above)
Comment (staltz) [reply test/compose.js:58]: Not sure why you're talking about throwing, I'm talking about asserting the returned value from `decompose`.

Comment (Powersource) [reply test/compose.js]: done

**Review by staltz — APPROVED** [review]
:tiger2:

**Pull request merged**